tests/nativeapp/test_post_deploy_for_app.py

```diff
@@ -120,19 +128,24 @@ def test_sql_scripts_with_no_warehouse_no_database(
     mock_conn_wh.return_value = None
     mock_conn_db.return_value = None
     mock_conn.return_value = MockConnectionCtx(None)
-    mock_cli_ctx.return_value = {
-        "ctx": {"native_app": {"name": "myapp"}, "env": {"foo": "bar"}}
-    }
-    with project_directory("napp_post_deploy") as project_dir:
-        processor = _get_run_processor(str(project_dir))
-
-        processor.execute_app_post_deploy_hooks()
+    with project_directory("napp_post_deploy_v2") as project_dir:
+        dm = DefinitionManager()
+        app_model: ApplicationEntityModel = dm.project_definition.entities["myapp"]
+        mock_cli_ctx.return_value = dm.template_context
+
+        execute_post_deploy_hooks(
+            console=cc,
+            project_root=dm.project_root,
+            post_deploy_hooks=app_model.meta.post_deploy,
+            deployed_object_type="application",
+            database_name=app_model.fqn.name,
+        )
```

Comment:
can we use the ApplicationEntity method `execute_post_deploy_hooks`?

Reply:
same in other places

Reply:
`ApplicationEntity.execute_post_deploy_hooks` inserts an extra `use_warehouse`, which we don't want to test here

Reply:
updated all the tests except one which is explicitly testing the no-warehouse behaviour